File: LeetCode/9. Palindrome Number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        n = str(x)
        left = 0 
        right = len(n) - 1
        while left < right:
            if(n[left] != n[right]):
                return False
            left += 1
            right -= 1
        return True 

# Example usage:
solution = Solution()
print(solution.isPalindrome(121))  # Output: True
print(solution.isPalindrome(-121)) # Output: False
print(solution.isPalindrome(10))   # Output: False

File: LeetCode/9. Palindrome Number.js
/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function(x) {
    let n = x.toString()
    let left  = 0 
    let right = n.length - 1
    while(left < right){
        if(n[left] != n[right]){
            return false
        }
        left += 1
        right -= 1

    }
    return true 
}

console.log(isPalindrome(121)) // true
console.log(isPalindrome(-121)) // false
console.log(isPalindrome(10)) // false


